src/main/java/io/scim2/swagger/client/ScimApiClient.java

```diff
@@ -758,23 +902,25 @@ public String escapeString(String str) {
      * Deserialize response body to Java object, according to the return type and
      * the Content-Type response header.
      *
-     * @param <T> Type
-     * @param response HTTP response
+     * @param <T>        Type
+     * @param response   HTTP response
      * @param returnType The type of the Java object
      * @return The deserialized Java object
      * @throws ScimApiException If fail to deserialize response body, i.e. cannot read response body
-     *   or the Content-Type of the response is not supported.
+     *                          or the Content-Type of the response is not supported.
      */
     @SuppressWarnings("unchecked")
-    public <T> T deserialize(Response response, Type returnType) throws ScimApiException {
+    public <T> T deserialize(HttpResponse response, Type returnType) throws ScimApiException {
```

Comment:
Changing the signature of the public method is not a good idea. Hope we are going to release a major version of scim?

Reply:
Yes a major version of the scim2 client will be released.